controllers/datadogagent/override/podtemplatespec_test.go

```diff
@@ -248,21 +308,74 @@ func TestPodTemplateSpec(t *testing.T) {
 				},
 			},
 			validateManager: func(t *testing.T, manager *fake.PodTemplateManagers) {
-				assert.Equal(t, "someregistry.com/a/b/c/cluster-agent:7.38.0-jmx", actualImage(manager, t))
+				assertImageConfigValues(
+					manager,
+					containerImageOptions{name: "someregistry.com/a/b/c/cluster-agent:7.38.0-jmx"},
+					t,
+				)
 			},
 		},
 		{
 			name: "15: given long URI, override name with slash, overrides name in current image",
 			existingManager: func() *fake.PodTemplateManagers {
-				return fakePodTemplateManagersWithImageOverride("someregistry.com/datadog/agent:9.99", t)
+				return fakePodTemplateManagersWithImageOverride(containerImageOptions{name: "someregistry.com/datadog/agent:9.99"}, t)
 			},
 			override: v2alpha1.DatadogAgentComponentOverride{
 				Image: &commonv1.AgentImageConfig{
 					Name: "otherregistry.com/agent",
 				},
 			},
 			validateManager: func(t *testing.T, manager *fake.PodTemplateManagers) {
-				assert.Equal(t, "someregistry.com/datadog/otherregistry.com/agent:9.99", actualImage(manager, t))
+				assertImageConfigValues(manager, containerImageOptions{name: "someregistry.com/datadog/otherregistry.com/agent:9.99"}, t)
+			},
+		},
+		{
+			name: "16: override image pull policy",
```

Comment:
16 and 17 are the new tests. the others are just refactored to use the new struct